File: ChatGPT/Direct_Attempt/c/output/output_04a.c
// ChatGPT-generated solution will go here.
#include <stdio.h>
#include <string.h>

#define SIZE 10
#define WORD "XMAS"
#define WORD_LEN 4

int check_word(char grid[SIZE][SIZE], int x, int y, int dx, int dy) {
    for (int i = 0; i < WORD_LEN; i++) {
        int nx = x + dx * i;
        int ny = y + dy * i;
        if (nx < 0 || nx >= SIZE || ny < 0 || ny >= SIZE || grid[nx][ny] != WORD[i]) {
            return 0;
        }
    }
    return 1;
}

int main() {
    char grid[SIZE][SIZE];
    FILE *f = fopen("input.txt", "r");
    int count = 0;

    for (int i = 0; i < SIZE; i++) {
        fscanf(f, "%s", grid[i]);
    }
    fclose(f);

    int directions[8][2] = {
        {0, 1}, {1, 0}, {1, 1}, {1, -1}, // horizontal, vertical, diagonal
        {0, -1}, {-1, 0}, {-1, -1}, {-1, 1} // reverse directions
    };

    for (int i = 0; i < SIZE; i++) {
        for (int j = 0; j < SIZE; j++) {
            for (int d = 0; d < 8; d++) {
                if (check_word(grid, i, j, directions[d][0], directions[d][1])) {
                    count++;
                }
            }
        }
    }

    printf("%d\n", count);
    return 0;
}

File: ChatGPT/Direct_Attempt/clojure/output/output_04a.clj
;; ChatGPT-generated solution will go here.
(ns output_04a
  (:require [clojure.string :as str]))

(def directions [[0 1] [1 0] [1 1] [-1 1] ; right, down, down-right, up-right
                 [0 -1] [-1 0] [-1 -1] [1 -1]]) ; left, up, up-left, down-left

(def target [\X \M \A \S])

(defn in-bounds? [grid [r c]]
  (and (>= r 0) (< r (count grid))
       (>= c 0) (< c (count (first grid)))))

(defn match-word? [grid r c [dr dc]]
  (every? identity
          (map-indexed
           (fn [i ch]
             (let [nr (+ r (* i dr))
                   nc (+ c (* i dc))]
               (and (in-bounds? grid [nr nc])
                    (= ch (get-in grid [nr nc])))))
           target)))

(defn part [lines]
  (let [grid (mapv vec lines)]
    (count
     (for [r (range (count grid))
           c (range (count (first grid)))
           dir directions
           :when (match-word? grid r c dir)]
       [r c dir]))))

(defn -main [& args]
  (let [filename (first args)
        lines (str/split-lines (slurp filename))]
    (println (part lines))))
